### DISCUSSION TITLE
How reset isDirty and isTouched on a specific field ?

### DISCUSSION THREAD

**Discussion by val2207** [react-hook-form/react-hook-form#6555] [Q&A] [answered]
I want when a field is not anymore visible reset the value with the default one and put isDirty and isTouched to false. 

Currently with reset function it's possible to change the value of isDirty and isTouched but for the whole form. With setValue you can put these two value at true but not at false. 

So there is any way to do that ?

**Comment by slugmandrew** [comment, marked as answer]
Hey! So `isDirty` and `isTouched` apply to the entire form, not individual elements. There is no way to 'get' the `isDirty` or `isTouched` for individual elements, so what is the benefit of being able to set them individually?

If you call `reset()` for a single field it will reset them both, as the props `keepDirty` and `keepTouched` suggest (just leave them as false):

![image](https://user-images.githubusercontent.com/153625/133635867-a7d8792d-cad9-4374-8a73-ef6860c9d550.png)

**Reply by slugmandrew** [reply]
@bluebill1049 See typo in docs above. PR here if that helps: https://github.com/react-hook-form/documentation/pull/666

**Reply by bluebill1049** [reply]
thanks very much @slugmandrew

**Comment by val2207** [comment]
This is not a blocker but let's suppose in this example (https://codesandbox.io/s/react-hook-form-js-forked-irnfd) that the error on the second field is displayed and when after display the error on the last field the visibility change from visible to not visible and back to visible (just write something on the first field) I want to reset just THIS last field not the second one with the error displayed. Currently this is not possible to reset the value and error displayed just for my last field because I can not edit isDirty and isTouched from this field.

Option A: Use reset and error on second field disappear. 
Option B: Use setValue (like inside this example) to reset the value but if an error is displayed for the last field this is not possible to remove this error. 

In both option it's not possible to do what I want

**Reply by slugmandrew** [reply]
Are you calling `reset` on the entire form? As in `reset()` instead of `reset(name)`? 

I would expect that to clear the error for the second field, but if you only reset the last field I don't see why that would reset any errors other than the ones that apply to that field (field 3).
